omnisafe/algorithms/model_based/cap.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2022 OmniSafe Team. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Implementation of the CAP algorithm. The CAP in safety-gym may unable to converge."""
+
+import numpy as np
+
+from omnisafe.algorithms import registry
+from omnisafe.algorithms.model_based.planner import CCEPlanner
+from omnisafe.algorithms.model_based.policy_gradient import PolicyGradientModelBased
+from omnisafe.common.lagrange import Lagrange
+from omnisafe.utils.config_utils import namedtuple2dict
+
+
+@registry.register
+class CAP(
+    PolicyGradientModelBased, CCEPlanner, Lagrange
+):  # pylint: disable=too-many-instance-attributes
+    """The Conservative and Adaptive Penalty (CAP) algorithm.
+
+    References:
+        Title: Conservative and Adaptive Penalty for Model-Based Safe Reinforcement Learning
+
+        Authors: Yecheng Jason Ma, Andrew Shen, Osbert Bastani, Dinesh Jayaraman.
+
+        URL: https://arxiv.org/abs/2112.07701
```

Comment:
```suggestion
    References:
        Title: Conservative and Adaptive Penalty for Model-Based Safe Reinforcement Learning
        Authors: Yecheng Jason Ma, Andrew Shen, Osbert Bastani, Dinesh Jayaraman.
        URL: https://arxiv.org/abs/2112.07701
```

Remove the empty lines in references.